lib/ddtrace/distributed_headers.rb

```diff
@@ -0,0 +1,66 @@
+require 'ddtrace/span'
+require 'ddtrace/ext/distributed'
+
+module Datadog
+  # DistributedHeaders provides easy access and validation to headers
+  class DistributedHeaders
+    include Ext::DistributedTracing
+
+    def initialize(env)
+      @env = env
+    end
+
+    def valid?
+      # Sampling priority is optional.
+      trace_id && parent_id
+    end
+
+    def trace_id
+      value = header(HTTP_HEADER_TRACE_ID).to_i
+      return if value <= 0 || value >= Span::MAX_ID
+      value
+    end
+
+    def parent_id
+      value = header(HTTP_HEADER_PARENT_ID).to_i
+      return if value <= 0 || value >= Span::MAX_ID
+      value
+    end
+
+    def sampling_priority
+      hdr = header(HTTP_HEADER_SAMPLING_PRIORITY)
+      # It's important to make a difference between no header,
+      # and a header defined to zero.
+      return unless hdr
+      value = hdr.to_i
+      return if value < 0
+      value
+    end
+
+    def self.inject!(span, env)
+      headers = { HTTP_HEADER_TRACE_ID => span.trace_id.to_s,
+                  HTTP_HEADER_PARENT_ID => span.span_id.to_s }
+      if span.sampling_priority
+        headers[HTTP_HEADER_SAMPLING_PRIORITY] = span.sampling_priority.to_s
+      end
+      env.merge! headers
+      env.delete(HTTP_HEADER_SAMPLING_PRIORITY) unless span.sampling_priority
+    end
+
+    def self.extract(env)
```

Comment:
More a design nitpick, but I'd prefer to do less things here and keep `DistributedHeaders` more like as presenter around the environment object. I think the concept of a `Context` should belong to somewhere else (probably to whoever call this).

Reply:
OK, my sense is that this probably, indeed, does not belong to something like `distributed_headers`, true enough. How about, then, renaming it to, say, `propagation/http` as in https://github.com/DataDog/dd-trace-py/blob/master/ddtrace/propagation/http.py Because, it's of general usage, it's proven useful in Python (extracting and injecting those values from headers to hash is quite a straightforward use case and many libraries may benefit from it, even if today only Faraday uses it).

Reply:
Generally I'd prefer to defer things like this until they're proven useful, but I'm pretty sure you can anticipate more use-cases and scenarios for it than I do, so I'm fine with it.